src/program.cpp

```diff
@@ -597,37 +277,16 @@ void program::from_value(const value& v)
         this->impl->ctx.from_value(v.at("context"));
     }
 
-    std::unordered_map<std::string, instruction_ref> instructions;
-    for(const value& node : v.at("nodes"))
+    auto val_modules = v.at("modules");
+    for(const auto& vv : val_modules)
     {
-        instruction_ref output;
-        auto name   = node.at("name").to<std::string>();
-        auto fields = node.at("operator");
-        if(name == "@param")
-        {
-            output = this->add_parameter(fields["parameter"].to<std::string>(),
-                                         migraphx::from_value<shape>(node.at("shape")));
-        }
-        else if(name == "@literal")
-        {
-            output = this->add_literal(migraphx::from_value<literal>(node.at("literal")));
-        }
-        else
-        {
-            auto op = make_op(name, fields);
-            std::vector<instruction_ref> inputs;
-            std::transform(node.at("inputs").begin(),
-                           node.at("inputs").end(),
-                           std::back_inserter(inputs),
-                           [&](const value& i) { return instructions[i.to<std::string>()]; });
-            if(name == "@return")
-                output = this->add_return(inputs);
-            else
-                output = this->add_instruction(op, inputs);
-        }
-        instructions[node.at("output").to<std::string>()] = output;
+        const auto& key = vv.get_key();
+        auto val        = vv.without_key();
+        module modl;
+        modl.from_value(val);
+        modl.finalize(this->impl->ctx);
```

Comment:
We shouldn't finalize the module until they are all loaded.

Reply:
fixed